recipes/build_recipe_macos.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# -----------------------------------------------------------------------------
+# Run this from the root of the OpenTurbine repository with the following arguments:
+# ./recipes/build_recipe_macos.sh <path_to_openturbine_root> <path_to_spack_root>
+# -----------------------------------------------------------------------------
+
+# Exit on error
+set -e
+
+# Clone OpenTurbine repository if not present in the provided path
+openturbine_path=$(dirname "$1") # Get the directory of the provided path
+if [ ! -d "$openturbine_path/openturbine" ]; then
+    git clone --recursive https://github.com/Exawind/openturbine.git $openturbine_path/openturbine
+fi
+cd $openturbine_path/openturbine
+
+# Install Spack if not present in the provided path
+spack_path=$(dirname "$2") # Get the directory of the provided path
+if [ ! -d "$spack_path/spack" ]; then
+    git clone https://github.com/spack/spack.git $spack_path/spack
+fi
+source $spack_path/spack/share/spack/setup-env.sh
+spack compiler find
+spack external find
+
+# Install required packages using Spack if not present
+install_if_missing() {
```

Comment:
I thought `spack install` would issue a similar message if the requested spec was already installed.  I don't think this wrapper should be necessary.

Reply:
It does not really issue any messages, so I added this wrapper just to make sure we're not duplicating anything there. However, it does not try to re-install it either if the package is found.